### PR TITLE
[DBParameterGroup]Soft fail for `DescribeDBParameterGroups` call

### PR DESCRIPTION
**Pull request by moataz-mhmd**
*Description of changes:*
Implementing soft failing while doing `DescribeDBPrameterGroups` call to be backward compatible with current CloudFormation customer to prevent `AccessDenied` exception for existing stacks.
Changes:
- Compose `DBParameterGroupArn` from `Request` and `ResourceModel` instead of retrieving it from `DescribeDBPrameterGroups` call.
- Remove `fetchDBParameterGroupArn` as it triggers `DescribeDBPrameterGroups` call.
- Apply `setDBParameterGroupArn` in all handlers.
- Use in progress soft fail error rule set while executing `DescribeDBPrameterGroups` call in read handler.
- Update relative unit tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java [hunk 1]
```diff
@@ -7,6 +7,13 @@
 @lombok.ToString
 @lombok.EqualsAndHashCode(callSuper = true)
 public class CallbackContext extends StdCallbackContext {
+    final String ARN_TEMPLATE = "arn:aws:rds:%s:%s:pg:%s";
+
     private boolean parametersApplied;
     private String dbParameterGroupArn;
```
Comment (osdrv) [inline aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java:13]: Let's ensure that `Arn` is a serializable datum

File: aws-rds-dbparametergroup/src/test/java/software/amazon/rds/dbparametergroup/UpdateHandlerTest.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public void handleRequest_SimpleSuccess() {
         assertThat(response.getMessage()).isNull();
         assertThat(response.getErrorCode()).isNull();
 
-        verify(proxyRdsClient.client(), times(2)).describeDBParameterGroups(any(DescribeDbParameterGroupsRequest.class));
+        verify(proxyRdsClient.client(), times(1)).describeDBParameterGroups(any(DescribeDbParameterGroupsRequest.class));
```
Comment (osdrv) [inline aws-rds-dbparametergroup/src/test/java/software/amazon/rds/dbparametergroup/UpdateHandlerTest.java:112]: Profit! :-D

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java [hunk 3]
```diff
@@ -71,6 +71,8 @@ public BaseHandlerStd(final HandlerConfig config) {
                                                                              final ResourceHandlerRequest<ResourceModel> request,
                                                                              final CallbackContext callbackContext,
                                                                              final Logger logger) {
+        callbackContext.setDbParameterGroupArn(request.getRegion(), request.getAwsAccountId(),
+                request.getDesiredResourceState().getDBParameterGroupName());
```
[the remark this answers is not in the document]
Comment (moataz-mhmd) [reply aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java]: Done.

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java (hunk 1 above)
Comment (moataz-mhmd) [reply aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/CallbackContext.java]: I kept it simple as String. We don't need any details back from Arn from it.

File: aws-rds-dbparametergroup/src/test/java/software/amazon/rds/dbparametergroup/UpdateHandlerTest.java (hunk 2 above)
Comment (moataz-mhmd) [reply aws-rds-dbparametergroup/src/test/java/software/amazon/rds/dbparametergroup/UpdateHandlerTest.java:112]: Yes 😀

File: aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/Translator.java [hunk 4]
```diff
@@ -131,6 +136,18 @@ static ResourceModel translateFromDBParameterGroup(final DBParameterGroup dbPara
                 .collect(Collectors.toList());
     }
 
+    static String buildParameterGroupArn(final ResourceHandlerRequest<ResourceModel> request) {
```
Comment (osdrv) [inline aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/Translator.java:139]: Let's please return an Arn and do a string conversion on the invocation side: it is always easier to convert it to string later on than to restore an Arn object.

Comment (moataz-mhmd) [reply aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/Translator.java:139]: Make sense.

**Review by osdrv — APPROVED** [review]

**Review by Outige — APPROVED** [review]

**Pull request merged**